src/middleware/language.js

```diff
@@ -4,30 +4,39 @@ const parser = require('accept-language-parser');
 const _ = require('lodash');
 const { languageCodeExists } = require('country-language');
 
+const DEFAULT_LANGUAGE = 'en';
+
 module.exports = setup;
 
 /**
- * Find matching language by partial match.
+ * Intersect the list of user accept languages over the list of authorized languages
+ * and return the first matching language
  *
- * To match, the language must have the same country code.
+ * iterate over the ordered list of user accepted languages and:
+ *   1/ find exact match (country + region)
+ *   2/ find partial match (country code)
  *
- * @param   {Array} languages Array of known languages.
- * @param   {Array} parsedLanguages Array of parsed languages from Accept-Language.
+ * @param   {Array} authorizedLanguages Array of authorized/known languages.
+ * @param   {Array} userAcceptLanguages Array of parsed languages from Accept-Language.
  * @returns {String|null} The first matching language if found, null otherwise.
  */
-function findPartialLanguageMatch(languages, parsedLanguages) {
-  const languagesByCode = _.reduce(languages, (languagesResult, language) => {
-    const code = language.split('-')[0];
-    languagesResult[code] = language;
-    return languagesResult;
-  }, {});
+function findLanguageMatch(authorizedLanguages, userAcceptLanguages) {
+  for (const language of userAcceptLanguages) {
+    // find exact matching
+    const foundExactLanguage = _.find(authorizedLanguages, (lang) => {
+      return lang === `${language.code}-${language.region}`;
```

Comment:
`language` does not necessarily have a region.

Reply:
But if the array is ordered, why can't you do it in one pass? Why do you need 2 finds ? let's say the user has `en,fr-FR;q=0.5`. The selected language here will be `fr-FR`. Don't you think it should be `en` ?

Reply:
`en,fr-FR` for the accept-language header and ['en', 'fr'] -> 'en'
`en-US,fr-FR` for the accept-language header and ['en', 'fr'] -> 'en'
-> two passes